src/codecs/hdr/encoder.rs

```diff
@@ -17,8 +44,24 @@ impl<W: Write> HdrEncoder<W> {
 
     /// Encodes the image ```data```
     /// that has dimensions ```width``` and ```height```
-    pub fn encode(mut self, data: &[Rgb<f32>], width: usize, height: usize) -> ImageResult<()> {
-        assert!(data.len() >= width * height);
+    pub fn encode(self, rgb: &[Rgb<f32>], width: usize, height: usize) -> ImageResult<()> {
+        self.encode_pixels(rgb.iter().map(|&rgb| to_rgbe8(rgb)), width, height)
+    }
+
+    /// Encodes the image ```data```
+    /// that has dimensions ```width``` and ```height```.
+    /// The callback must return the color for the given flattened index of the pixel (row major).
+    pub fn encode_pixels(
+        mut self,
+        mut flattened_rgbe_pixels: impl ExactSizeIterator<Item = Rgbe8Pixel>,
```

Comment:
i needed to add a function that uses an iterator, because the byte buffer might not be aligned, so we just read individual values from the buffer with bytemuck in an iterator

a future performance improvement might be to check whether the buffer is aligned, and then cast the whole thing (which will be the case nearly all the time)